java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/ClientCredentialsWithJWTAssertionGrantTokenRequestBuilder.java

```diff
@@ -50,7 +51,7 @@ enum ConfigProperty implements ConfigService.ConfigProperty {
     // 'client_credentials' is MSFT
     //for Google, this is "urn:ietf:params:oauth:grant-type:jwt-bearer"
     @Getter(onMethod_ = @Override)
-    private final String grantType = "client_credentials";
+    private final String grantType = "client_credentials_jwt_assertion";
```

Comment:
This will break compatibility earlier terraform modules of this `0.4.x` generation, so we can't do it.

I think it would be better to parameterize the type of credential used, if needed, as a different config variable. (or duck-type it based on what sources of credentials are configured)

Reply:
TOKEN_REQUEST_STRATEGY, or something? (where it defaults based on `GRANT_TYPE`, but if you pass explicit value you can override it?)